vmm/src/serial_manager.rs

```diff
@@ -289,12 +289,13 @@ impl SerialManager {
                                     // be considered as a regular error. Instead it is more
                                     // appropriate to retry, by calling into epoll_wait().
                                     continue;
+                                } else {
+                                    return Err(Error::Epoll(e));
                                 }
-                                return Err(Error::Epoll(e));
                             }
                         };
 
-                        if matches!(in_file, ConsoleOutput::Socket(_)) && num_events == 0 {
+                        if matches!(in_file, ConsoleOutput::Pty(_)) && num_events == 0 {
```

Comment:
Looks like this was inadvertently changed in this commit - 287887c99c98e47819b3ac999cf4fa94c98407d5
```
-                        if mode != ConsoleOutputMode::Socket && num_events == 0 {
+                        if matches!(in_file, ConsoleOutput::Socket(_)) && num_events == 0 {
```
@alyssais Do you think @praveen-pk's change here to check for Pty makes sense?

Reply:
@praveen-pk Logically - I think the change above should be in one commit and the change below in another.

Reply:
restructured the patches, looks cleaner now.